src/Workspaces/Core/Portable/Options/IGlobalOptionService.cs

```diff
@@ -41,5 +41,10 @@ internal interface IGlobalOptionService
         IEnumerable<IOption> GetRegisteredOptions();
 
         event EventHandler<OptionChangedEventArgs> OptionChanged;
+
+        /// <summary>
+        /// Refreshes the stored value of a serialized option. This should only be called from serializers.
+        /// </summary>
+        void RefreshOption(OptionKey optionKey, object newValue);
```

Comment:
There's a deeper piece of work to detangle this but I didn't want to attack that now. I think the better interface to give some sort of callback to the serializer, but I'm not sure what that looks like.